### ISSUE TITLE
Question: Why is btls included when using Andriod native client and minimun target is apk 21?

### ISSUE THREAD

**Issue by bentmar** [xamarin/xamarin-android#2573]
I get more and more confused reading docs about httpclient implementation and tls.

In the microsoft docs it says that using the androidClientHandler reduces the size of the apk (since it will not use btls for tls 1.2) => only works on api >=21 

And on another doc AndroidClientHandler now defaults to use btls becuase of the support for httpwebrequest with tls 1.2...

And what ever i do, I allways find that my apk ends up with the btls lib.. 

according to the docs https://docs.microsoft.com/en-us/xamarin/android/app-fundamentals/http-stack?tabs=windows The androidClientHandler is a good choice because its fast/native and reduces app size(becuase it doesnt use btls) -- Then please tell me how to get an apk without btls! Very confusing..

They also say that the managed httpclient doestn support tls 1.2, to further down in the docs say that i DOES! because it is using btls. The confusion is real!!

I do not want to use btls when building apps that dont use webclient and has the minimun target to api 21.. 

Im on the latest version of vs2017

Thx! :)

**Comment by erikpowa** [comment]
> In the microsoft docs it says that using the androidClientHandler reduces the size of the apk (since it will not use btls for tls 1.2) => only works on api >=21
```xml
android:minSdkVersion="21"
<AndroidHttpClientHandlerType>Xamarin.Android.Net.AndroidClientHandler</AndroidHttpClientHandlerType>
<AndroidTlsProvider>legacy</AndroidTlsProvider>
```

It's working for me.



> And on another doc AndroidClientHandler now defaults to use btls becuase of the support for httpwebrequest with tls 1.2...

It says that the `AndroidTlsProvider` property is NOT set to `legacy` therefore, by default, btls is being used to support TLS 1.2 across all API.
```xml
<AndroidTlsProvider />
```

> And what ever i do, I allways find that my apk ends up with the btls lib..

You may have a custom environment that overrides the properties.

**Comment by JonDouglas** [comment]
@bentmar Apologies in advance. The documentation may be a little outdated and confusing. I would recommend adding a feedback item for the documentation here: https://docs.microsoft.com/en-us/xamarin/android/app-fundamentals/http-stack?tabs=windows#feedback

The default is `btls`, which was made in this commit: https://github.com/xamarin/xamarin-android/commit/edfa864b7a1f8cdcf81113c9e163bc56fafe2425

The best way to control this is by using the respective `$(AndroidTlsProvider)` property in your .csproj as @erikpowa mentions. You can see all `AndroidTlsProvider` values here: https://docs.microsoft.com/en-us/xamarin/android/deploy-test/building-apps/build-process#packaging-properties

If you are using `AndroidClientHandler`, there is no need for `btls` unless you are still using `WebClient` or have to support TLS 1.2+ on < API 21 devices.

**Comment by JonDouglas** [comment]
I'm closing this issue as it seems like you are unblocked. Please feel free to open a new issue or reopen this issue if this is still an issue!

**Comment by jvreeker** [comment]
I now use the zerossl certificate, for now everything is working again. Maybe in the future I will go back to letsencrypt.
